File: project4/src/People/Person.java
package People;

public class Person {
    private String name;
    private int floorNumber;
    private int classNumber;
    private Boolean inside;
    private int fatige;
    private int tiredness;
    
    public Person(String namePerson,int floorNumberPerson,int classNumberPerson,int tirednessFactor){
        name=namePerson;
        floorNumber=floorNumberPerson;
        classNumber=classNumberPerson;
        tiredness=tirednessFactor;
        inside=false;
        fatige=0;
    }

    public String getName(){
        return name;
    }
    public int getClassFloor(){
        return floorNumber;
    }
    public int getClassNumber(){
        return classNumber;
    }
    public int getFatige(){
        return fatige;
    }
    public Boolean getInside(){
        return inside;
    }
    public void setFatige(int newFatige){
        fatige=newFatige;
    }
    public void addFatige(int newFatige){
        fatige+=newFatige;
    }
    public void setInside(Boolean newInside){
        inside=newInside;
    }

    public void print(){
        System.out.println("Name: "+name);
        System.out.println("Class Number: "+Integer.toString(classNumber)+" Floor: "+Integer.toString(floorNumber));
        System.out.println("Fatige: "+Integer.toString(fatige));
        System.out.println("Inside: "+Boolean.toString(inside));
    }  

    public void printName(){
        System.out.println("Name: "+name);
        System.out.println("Fatige: "+Integer.toString(fatige));
    }

    public void operate(int hours){
        addFatige(hours*tiredness);
    }
}

File: project3/modules/person.cpp
#include <iostream>

#include <person.h>
// #include "../include/person.h"

Person::Person(const std::string namePerson, const unsigned int floorClass ,const unsigned int numberClass,const unsigned int tirednessPerson):
floor(floorClass),number(numberClass),inside(false),fatige(0),tiredness(tirednessPerson){
    name=namePerson;
    print();
}

Person::~Person(){
    print();
}

const std::string& Person::getName() const{
    return name;
}

void Person::getClass(int* floorClass,int* numberClass) const{
    *floorClass=floor;
    *numberClass=number;
}

unsigned int Person::getFatige() const{
    return fatige;
}

bool Person::getInside() const{
    return inside;
}

void Person::setFatige(int addFatige){
    fatige+=addFatige;
}

void Person::setInside(bool newInside){
    inside=newInside;
}

void Person::print() const{
    std::cout<<"Name: "<<name<<" ";
    std::cout<<"His Class floor: "<<floor<<" number:"<<number<<"\n";
    std::cout<<"Inside: "<<inside<<" fatige:"<<fatige<<"\n";
}

void Person::printName() const{
    std::cout<<"Name: "<<name<<"\n";
    std::cout<<"Fatige: "<<fatige<<"\n";
}

void Person::printEnter(std::string roomName,int index) const{
    if(index!=-1)
        std::cout<<name<<" enters "<<roomName <<index<<"\n";
    else
        std::cout<<name<<" enters "<<roomName<<"\n";
}

void Person::Person::printExit(std::string roomName) const{
    std::cout<<name<<" exits "<<roomName <<"\n";
}

void Person::operate(int hours){
    setFatige(tiredness*hours);
}